### ISSUE TITLE
Disable custom memory manager (Custom StlAllocator)

### ISSUE THREAD

**Issue by OnlyOneMoreGeek** [aws/aws-iot-device-sdk-cpp-v2#382]
Confirm by changing [ ] to [x] below:
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/) and didn't find any solution

**Known Issue**
- [ ] I'm using ATS data type endpoint: the endpoint should look like `<prefix>-ats.iot.<region>.amazonaws.com`

**Platform/OS/Device**
What are you running the sdk on?
Linux x86
**Describe the question**

When using the sdk, the Aws::Crt types that wrap stl containers use custom allocators by default. This disables the castings of the stl to the AWS::crt (e.g. std::string to Aws::Crt:String).

Is there any way to disable these custom allocators?

**Comment by jmklix** [comment]
I don't think that its possible to disable the custom memory manager for this sdk. Would using [user defined conversion functions](https://en.cppreference.com/w/cpp/language/cast_operator) work for your use case?

**Comment by OnlyOneMoreGeek** [comment]
> I don't think that its possible to disable the custom memory manager for this sdk. Would using [user defined conversion functions](https://en.cppreference.com/w/cpp/language/cast_operator) work for your use case?

I think the library should provide user-defined conversions for Aws::Crt::String and Aws:_Crt::Vector to those in the STL. I think it is necessary for the library to allow avoiding unnecessary copies. The other alternative should be to completely disable the memory manager as is done in the aws C++ sdk [Link](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/memory-management.html)

**Comment by jmklix** [comment]
Thanks for your feedback. This is something that is not currently supported, but I will change this to a feature request. I don't have a timeline for when this might get added, but I encourage you to upvote the initial comment if you would like this added to this sdk.
